luomu-libpcap/src/functions.rs

```diff
@@ -572,6 +571,15 @@ fn check_pcap_error(pcap_t: &PcapT, ret: i32) -> Result<()> {
         );
     }
 
+    // These two values don't care about the interface name, and success or
+    // error is checked for all the time. Thus handle now and return fast before
+    // doing the interface String copying business.
+    match ret {
+        PCAP_SUCCESS => return Ok(()),
+        PCAP_ERROR => return Err(get_error(pcap_t)?),
+        _ => (),
+    }
+
```

Comment:
How about adding `PcapT::get_ifname()` method which could be called in cases where interface name is needed. That way we would not need two `match` blocks and that would make this method a bit clearer since all error coders are handled in same place.